Spatial_Engine/Query/Geometry.cs

```diff
@@ -24,13 +24,43 @@
 using BH.oM.Geometry;
 using BH.oM.Base.Attributes;
 using BH.oM.Spatial.SettingOut;
+using BH.oM.Spatial.ShapeProfiles;
 using BH.oM.Base.Attributes;
 using System.ComponentModel;
+using System.Linq;
 
 namespace BH.Engine.Spatial
 {
     public static partial class Query
     {
+        /******************************************/
+        /****  Public Methods                  ****/
+        /******************************************/
+
+        [PreviousVersion("5.1", "BH.Engine.Architecture.Query.Geometry(BH.oM.Architecture.Elements.Grid)")]
+        [PreviousVersion("5.1", "BH.Engine.Architecture.Query.Geometry(BH.oM.Spatial.SettingOut.Grid)")]
+        [PreviousVersion("5.1", "BH.Engine.Geometry.Query.Geometry(BH.oM.Spatial.SettingOut.Grid)")]
+        [Description("Queries the defining Curve from the grid.")]
+        [Input("grid", "The Grid to get the defining curve from.")]
+        [Output("curve", "The Grids base geometrical curve object.")]
+        public static ICurve Geometry(this Grid grid)
+        {
+            return grid?.Curve;
+        }
+
+        /******************************************/
+
+        [Description("Queries edge curves from an IProfile.")]
+        [Input("profile", "The profile to get the geometry from.")]
+        [Output("edges", "The edge curves of the profile as a CompositeGeometry.")]
+        public static CompositeGeometry Geometry(this IProfile profile)
+        {
+            if (profile?.Edges == null)
+                return new CompositeGeometry();
```

Comment:
How useful is it to have this returned if the edges were null? Would a warning or error to the user be of any use? (Asking as I don't know the use case, so happy to be told it's not of use.)

Reply:
Was thinking of doing it, but opted not to, as this get called by the output of various components. hence this would raise additional potential warnings for outputs of components that might just be confusing things (such as the actual thing the method was actually trying to solve.). Happy to change to a `null` return instead though if you think that is better. Might actually be better than an empty Composite geometry

Reply:
I don't have much preference, I'm happy either way which is why I raised this as a comment rather than a request for changes. I am happy to approve what I'm seeing, just wanted to double check what thoughts you'd had on it?

Reply:
Has been changed to `null`, marking as resolved.